src/main/java/io/github/moulberry/notenoughupdates/profileviewer/level/task/SlayingTaskLevel.java

```diff
@@ -0,0 +1,288 @@
+			for (String slayer : slayers) {
+				ProfileViewer.Level level = skyblockInfo.get(slayer);
+				for (int i = 0; i < (int) level.level; i++) {
+					int asInt = slayerLevelUpXp.get(i).getAsInt();
+					sbXpGainedSlayer += asInt;
+				}
+			}
+		}
+
+		JsonObject bossCollectionsXp = slayingTask.getAsJsonObject("boss_collections_xp");
+
+		HashMap<String, Double> allComps = new HashMap<>();
+		JsonElement normalCompletions = Utils
+			.getElement(object, "dungeons.dungeon_types.catacombs.tier_completions");
+
+		JsonElement masterCompletions = Utils
+			.getElement(object, "dungeons.dungeon_types.master_catacombs.tier_completions");
+
+		if (normalCompletions != null) {
+			HashMap<String, Double> normalCompMap = NotEnoughUpdates.INSTANCE.manager.gson.fromJson(
+				normalCompletions.getAsJsonObject(),
+				HashMap.class
+			);
+			normalCompMap.forEach((floor, value) -> {
+				if (allComps.containsKey(floor)) {
+					allComps.put(floor, allComps.get(floor) + value);
+				} else {
+					allComps.put(floor, value);
+				}
+			});
+		}
+		if (masterCompletions != null) {
+			HashMap<String, Double> masterCompMap = NotEnoughUpdates.INSTANCE.manager.gson.fromJson(
+				masterCompletions.getAsJsonObject(),
+				HashMap.class
+			);
+
+			masterCompMap.forEach((floor, value) -> {
+				if (allComps.containsKey(floor)) {
+					allComps.put(floor, allComps.get(floor) + value);
+				} else {
+					allComps.put(floor, value);
+				}
+			});
+		}
+		int[] bossLow = {25, 50, 100, 150, 250, 1000};
+		int[] thorn = {25, 50, 150, 250, 400, 1000};
+		int[] bossHigh = {50, 100, 150, 250, 500, 750, 1000};
+		// THIS SERVER IS AWESOME I LOVE CONSISTENCY!!!!!!!
+
+		int bossCollectionXp = 0;
+		JsonArray dungeonCollectionXp = bossCollectionsXp.getAsJsonArray("dungeon_collection_xp");
+		for (int i = 1; i <= 7; i++) {
+			if (!allComps.containsKey(i + "")) continue;
+			double value = allComps.get(i + "");
+			switch (i) {
+				case 1:
+				case 2:
+				case 3:
+					bossCollectionXp += loopThroughCollection(
+						bossLow,
+						value,
+						dungeonCollectionXp
+					);
+					break;
+				case 4:
+					bossCollectionXp += loopThroughCollection(
+						thorn,
+						value,
+						dungeonCollectionXp
+					);
+					break;
+				case 5:
+				case 6:
+				case 7:
+					bossCollectionXp += loopThroughCollection(
+						bossHigh,
+						value,
+						dungeonCollectionXp
+					);
+					break;
+			}
+		}
+
+		JsonArray defeatKuudraXp = slayingTask.get("defeat_kuudra_xp").getAsJsonArray();
+		// kuudra
+
+		int sbXpDefeatKuudra = 0;
+
+		int kuudraBossCollection = 0;
+		if (object.has("nether_island_player_data")) {
+			JsonObject jsonObject = object.getAsJsonObject("nether_island_player_data").getAsJsonObject(
+				"kuudra_completed_tiers");
+			for (Map.Entry<String, JsonElement> stringJsonElementEntry : jsonObject.entrySet()) {
+				String key = stringJsonElementEntry.getKey();
+				int value = stringJsonElementEntry.getValue().getAsInt();
+				if (key.equals("none")) {
```

Comment:
Look at the existing KuudraPage. We surely have an list for the tier names already you can use inside an for loop to prevent code duplication.

Reply:
Its in CrimsonIslePage L155, but will do it myself since

Reply:
Done.